shm/src/Map.cpp

```diff
@@ -0,0 +1,45 @@
+#include "Map.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <stdexcept>
+#include <exception>
+
+constexpr int COORDINATE_MIN = 0;
+constexpr int COORDINATE_MAX = 99;
+constexpr int ISLANDS_COUNT = 10;
+constexpr int TRIALS = ISLANDS_COUNT * 10;
+
+Map::Map() {
+    std::random_device rd;
+    std::mt19937 generator(rd);
+    std::uniform_int_distribution<> distribution{
+        COORDINATE_MIN, COORDINATE_MAX
+    };
+    int count{};
+    while (vectorOfIslands_.size() != ISLANDS_COUNT || count < TRIALS) {
```

Comment:
I don't undestand what kind role have count variable.

Reply:
It may be redundant if the class is properly used. The idea was to avoid the infinite loop when the random generator is unable to generate unique coordinates for 10 islands, i.e. let's say coordinate_min is 0 and coordinate_max is 2, thus there is not enough possible combinations of coordinates to create 10 islands - with count variable the loop ends after 100  of ineffective tries.